torch/onnx/_internal/fx/dynamo_graph_extractor.py

```diff
@@ -40,7 +40,12 @@ def __init__(self):
 
     def __enter__(self):
         for class_type, (flatten_func, unflatten_func) in self._extensions.items():
-            pytree._register_pytree_node(class_type, flatten_func, unflatten_func)
+            pytree._register_pytree_node(
+                class_type,
+                flatten_func,
+                unflatten_func,
+                _register_cxx_pytree_node=False,
```

Comment:
Why is this false?

Reply:
`transformers.modeling_outputs.ModelOutput` is the only type to call this function. That type is an abstract type and will never used in real use cases.

- https://github.com/huggingface/transformers/pull/26618#issuecomment-1772310393

> * `ModelOutput` is not designed to be used directly (well, to be fair, within `transformers` code base).
>   
>   * All usage should inheritate `ModelOutput` and specify the attributes an output have), see, for example, https://github.com/huggingface/transformers/blob/929134bf65ac986c12c423c30b0db8a239f3b195/src/transformers/modeling_outputs.py#L25-L47
>   * (it's kind abstract class, despite we can still use it directly, but that usage is outside `transformers` scope.)
>   * Moreover, I would say it is super rare users will use `ModelOutput` directly  (in a manually way) to store their model output (without creating a subclass first).
>     
>     * (and even more rare, this usage along with `PyTreeExtensionContext` context manager)

------

I added this `_register_cxx_pytree_node=False` because the context manager will unregister the types (currently only `ModelOutput`) when exiting the context. In the Python pytree, it pops out types in `_pytree.SUPPORTED_NODE`. But in C++ Python, we do not offer an API to delete the node registration. On the next time we enter the context manager, the C++ pytree will raise an error for duplicate registration for `ModelOutput`.

We need to discuss how we design the node registry APIs for deleting/updating the registry.